### PR TITLE
Fix compilation errors with cce 17.0.1

### PR DESCRIPTION
**Pull request by raguridan-bsc**

### CODE REVIEW

**Comment by FussyDuck** [comment]
[CLA assistant check](https://bol-claassistant.ecmwf.int:443/ecmwf-ifs/ectrans?pullRequest=173) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://bol-claassistant.ecmwf.int:443/ecmwf-ifs/ectrans?pullRequest=173) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://bol-claassistant.ecmwf.int:443/check/ecmwf-ifs/ectrans?pullRequest=173) it.</sub>

File: src/trans/gpu/CMakeLists.txt [hunk 1]
```diff
@@ -114,6 +114,10 @@ foreach( prec dp sp )
           target_compile_definitions( trans_gpu_${prec} PRIVATE TRANS_SINGLE PARKINDTRANS_SINGLE )
         endif()
 
+        if( ROCM6 AND CMAKE_Fortran_COMPILER_ID MATCHES Cray )
```
Comment (samhatfield) [inline src/trans/gpu/CMakeLists.txt:117]: Is there some missing logic here to set the CMake variable `ROCM6`?

Comment (raguridan-bsc) [reply src/trans/gpu/CMakeLists.txt:117]: The ROCM6 variable is set in the bundle toolchain file for cce 17.0.1. 
Not sure if it's possible to detect that we are using ROCM6.

**Comment by raguridan-bsc** [comment]
Right.

If we are not required to support rocm < 6, we can probably drop the whole PR, since there was a previous PR updating the include paths for rocm 6.

**Comment by samhatfield** [comment]
That PR only affects the develop branch, not this one. Let me make some suggested changes which will hard require ROCM >= 6.

**Comment by samhatfield** [comment]
Actually, too complicated. If you could just reopen the PR but with the ROCm >= 6 version only kept (no changes to CMakeLists.txt required), that would be great.

**Comment by raguridan-bsc** [comment]
The changes for supporting only rocm 6 are already on the redgreengpu branch: https://github.com/ecmwf-ifs/ectrans/commit/7bdd751050b246ca6b22c4d446d35cad3716467a

I missed the commit, mainly because I was using the bundle tag for ecTrans.

**Comment by samhatfield** [comment]
I completely forgot about that. Okay I'll close this then.

**Pull request closed without merging**